mithril-stm/Cargo.toml

```diff
@@ -19,15 +19,23 @@ rug-backend = ["rug/default"]
 num-integer-backend = ["num-bigint", "num-rational", "num-traits"]
 benchmark-internals = [] # For benchmarking multi_sig
 future_proof_system = [] # For activating future proof systems
+future_snark = [] # For activating snark features
 
 [dependencies]
+anyhow.workspace = true
 blake2 = "0.10.6"
 # Enforce blst portable feature for runtime detection of Intel ADX instruction set.
 blst = { version = "0.3.16", features = ["portable"] }
 digest = { workspace = true }
+ff = "0.13.1"
+group = "0.13.0"
+midnight-circuits = { git = "https://github.com/midnightntwrk/midnight-zk", rev = "c88a50c2169f060120a52ad0980de90f08bc9535" }
```

Comment:
These dependencies should be used only if the `future_snark` feature is activated.

Also a very important point: we can't merge the PR until the midnight dependencies are published on crates.io.